configs/config.go

```diff
@@ -16,14 +18,41 @@ type API struct {
 }
 
 type Terraform struct {
-	Bin    *string
-	Source *string
+	Bin            *string
+	Source         *string
+	TFInitOptions  *string `yaml:"tfInitOptions,omitempty"`
+	TFApplyOptions *string `yaml:"tfApplyOptions,omitempty"`
 }
 
 type DeployConfig struct {
 	UseCached       bool
 	DeployLocalPath string
 	BatchMode       bool
+	TFInitOptions   string
+	TFApplyOptions  string
 }
 
 var Regions = []string{"us-east-1", "us-east-2", "us-west-1", "us-west-2"}
+
+// Coalesce returns the first non-empty vluae, but takes into account a loading order,
+// which is CLI args > config file > environment variables > some default
+func Coalesce(arg *string, config *string, envvar *string, def *string) *string {
```

Comment:
I'm more familiar with:
`environment variables > CLI args > config file >  some default`

For example, this is what the AWS CLl does with `AWS_REGION`/`--profile`/ value in `~/.aws/config`

Reply:
wait, I'm confused.
It's definitely `CLI Arg > env var > config file`

Which makes sense intuitively, as it `user supplied options > current execution context > persistent system configuration`

Reply:
I have this changed now to this:

> It's definitely CLI Arg > env var > config file